utils/utils_test.go

```diff
@@ -167,3 +169,85 @@ func TestIsWildcardParentheses(t *testing.T) {
 		t.Errorf("TestIsWildcardParentheses() == %t, want %t", got, want)
 	}
 }
+
+func TestAntPathToRegExp(t *testing.T) {
+	var fileSystemPaths []string = []string{
+		"dev/a/b.txt",
+		"dev/a/bb.txt",
+		"dev/a/bc.txt",
+		"dev/aa/b.txt",
+		"dev/aa/bb.txt",
+		"dev/aa/bc.txt",
+		"dev/aa/b.zip",
+		"dev/aa/b.zip",
+		"dev/aa/bc.zip",
+		"dev/a1/a2/a3/b.txt",
+		"dev/a1/a2/b.txt",
+
+		"test/a/b.txt",
+		"test/a/bb.txt",
+		"test/a/bc.txt",
+		"test/aa/b.txt",
+		"test/aa/bb.txt",
+		"test/aa/bc.txt",
+		"test/aa/b.zip",
+		"test/aa/b.zip",
+		"test/aa/bc.zip",
+	}
+	tests := []struct {
+		name               string
+		antPattern         string
+		expectedRegExp     string
+		allFileSystemPaths []string
+		matchedPaths       []string
+	}{
+		{"check '?' in file's name", "dev/a/b?.txt", addRegExpPrefixAndSuffix("dev/a/b.{1}\\.txt"), fileSystemPaths, []string{"dev/a/bb.txt", "dev/a/bc.txt"}},
+		{"check '?' in directory's name", "dev/a?/b.txt", addRegExpPrefixAndSuffix("dev/a.{1}/b\\.txt"), fileSystemPaths, []string{"dev/aa/b.txt"}},
+		{"check '*' in file's name", "dev/a/b*.txt", addRegExpPrefixAndSuffix("dev/a/b([^/]*)\\.txt"), fileSystemPaths, []string{"dev/a/b.txt", "dev/a/bb.txt", "dev/a/bc.txt"}},
+		{"check '*' in directory's name", "dev/*/b.txt", addRegExpPrefixAndSuffix("dev/([^/]*)/b\\.txt"), fileSystemPaths, []string{"dev/a/b.txt", "dev/aa/b.txt"}},
+		{"check '**' in directory path", "**/b.txt", addRegExpPrefixAndSuffix("(.*/)?b\\.txt"), fileSystemPaths, []string{"dev/a/b.txt", "dev/aa/b.txt", "test/a/b.txt", "test/aa/b.txt", "dev/a1/a2/a3/b.txt", "dev/a1/a2/b.txt"}},
+		{"combine all signs", "**/b?.*", addRegExpPrefixAndSuffix("(.*/)?b.{1}\\.([^/]*)"), fileSystemPaths, []string{"dev/a/bb.txt", "dev/a/bc.txt", "dev/aa/bb.txt", "dev/aa/bc.txt", "dev/aa/bc.zip", "dev/a1/a2/a3/b.txt", "dev/a1/a2/b.txt", "test/a/bb.txt", "test/a/bc.txt", "test/aa/bb.txt", "test/aa/bc.txt", "test/aa/bc.zip"}},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			regExpStr := antPatternToRegExp(test.antPattern)
+			if test.expectedRegExp != regExpStr {
+				t.Error("Unmatched! for ant pattern `" + test.antPattern + "` : Expected `" + test.expectedRegExp + "` Got `" + regExpStr + "`")
+			}
+			var matches []string
+			for _, checkedPath := range fileSystemPaths {
+				match, _ := regexp.MatchString(regExpStr, checkedPath)
+				if match {
+					matches = append(matches, checkedPath)
+				}
+			}
+			if !equalSlicesIgnoreOrder(matches, test.matchedPaths) {
+				t.Error("Unmatched! : ant pattern `" + test.antPattern + "` matches paths:\n[" + strings.Join(test.matchedPaths, ",") + "]\nbut got:\n[" + strings.Join(matches, ",") + "]")
+			}
+		})
+	}
+}
+
+func addRegExpPrefixAndSuffix(str string) string {
+	return "^" + str + "$"
+}
+
+func equalSlicesIgnoreOrder(s1, s2 []string) bool {
```

Comment:
I think we can simplify this function, by sorting first, and then comparing the two slices.
For additional performance, we can compare the size of the slices, before sorting. If the sizes do not match, return false immediately.